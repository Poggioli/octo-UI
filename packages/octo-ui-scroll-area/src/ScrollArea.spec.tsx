import { render } from "@testing-library/react";
import {
  ScrollArea,
  ScrollAreaCorner,
  ScrollAreaScrollbar,
  ScrollAreaThumb,
  ScrollAreaViewport
} from "./ScrollArea";

describe("<ScrollArea />", () => {
  it(`GIVEN a <ScrollArea />
      WHEN render is called
      THEN should render a <ScrollArea />`, () => {
    const rendered = render(
      <ScrollArea type="always">
        <ScrollAreaViewport>
          {Array.from({ length: 50 }).map((_, i, a) => (
            <p key={`v1.2.0-beta.${a.length - i}`}>{`v1.2.0-beta.${
              a.length - i
            }`}</p>
          ))}
        </ScrollAreaViewport>
        <ScrollAreaScrollbar orientation="vertical">
          <ScrollAreaThumb />
        </ScrollAreaScrollbar>
        <ScrollAreaScrollbar orientation="horizontal">
          <ScrollAreaThumb />
        </ScrollAreaScrollbar>
        <ScrollAreaCorner />
      </ScrollArea>
    );
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <ScrollArea />
      WHEN toString is called
      THEN return a string`, () => {
    expect(ScrollArea.toString()).toBe(".octo-UI-c-ipNenM");
  });

  it(`GIVEN a <ScrollAreaViewport />
      WHEN toString is called
      THEN return a string`, () => {
    expect(ScrollAreaViewport.toString()).toBe(".octo-UI-c-CnFzT");
  });

  it(`GIVEN a <ScrollAreaScrollbar />
      WHEN toString is called
      THEN return a string`, () => {
    expect(ScrollAreaScrollbar.toString()).toBe(".octo-UI-c-dDCVUb");
  });

  it(`GIVEN a <ScrollAreaThumb />
      WHEN toString is called
      THEN return a string`, () => {
    expect(ScrollAreaThumb.toString()).toBe(".octo-UI-c-hTfRLm");
  });

  it(`GIVEN a <ScrollAreaCorner />
      WHEN toString is called
      THEN return a string`, () => {
    expect(ScrollAreaCorner.toString()).toBe(".octo-UI-c-jMaIDO");
  });
});
