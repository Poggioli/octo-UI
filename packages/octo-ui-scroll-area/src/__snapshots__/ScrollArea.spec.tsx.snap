// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrollArea /> GIVEN a <ScrollArea />
      WHEN render is called
      THEN should render a <ScrollArea /> 1`] = `
<div>
  <div
    class="octo-UI-c-ipNenM octo-UI-c-ipNenM-dQEJNe-color-unstyled"
    dir="ltr"
    style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
  >
    <style>
      [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
    </style>
    <div
      class="octo-UI-c-CnFzT"
      data-radix-scroll-area-viewport=""
      style="overflow-x: scroll; overflow-y: scroll;"
    >
      <div
        style="min-width: 100%; display: table;"
      >
        <p>
          v1.2.0-beta.50
        </p>
        <p>
          v1.2.0-beta.49
        </p>
        <p>
          v1.2.0-beta.48
        </p>
        <p>
          v1.2.0-beta.47
        </p>
        <p>
          v1.2.0-beta.46
        </p>
        <p>
          v1.2.0-beta.45
        </p>
        <p>
          v1.2.0-beta.44
        </p>
        <p>
          v1.2.0-beta.43
        </p>
        <p>
          v1.2.0-beta.42
        </p>
        <p>
          v1.2.0-beta.41
        </p>
        <p>
          v1.2.0-beta.40
        </p>
        <p>
          v1.2.0-beta.39
        </p>
        <p>
          v1.2.0-beta.38
        </p>
        <p>
          v1.2.0-beta.37
        </p>
        <p>
          v1.2.0-beta.36
        </p>
        <p>
          v1.2.0-beta.35
        </p>
        <p>
          v1.2.0-beta.34
        </p>
        <p>
          v1.2.0-beta.33
        </p>
        <p>
          v1.2.0-beta.32
        </p>
        <p>
          v1.2.0-beta.31
        </p>
        <p>
          v1.2.0-beta.30
        </p>
        <p>
          v1.2.0-beta.29
        </p>
        <p>
          v1.2.0-beta.28
        </p>
        <p>
          v1.2.0-beta.27
        </p>
        <p>
          v1.2.0-beta.26
        </p>
        <p>
          v1.2.0-beta.25
        </p>
        <p>
          v1.2.0-beta.24
        </p>
        <p>
          v1.2.0-beta.23
        </p>
        <p>
          v1.2.0-beta.22
        </p>
        <p>
          v1.2.0-beta.21
        </p>
        <p>
          v1.2.0-beta.20
        </p>
        <p>
          v1.2.0-beta.19
        </p>
        <p>
          v1.2.0-beta.18
        </p>
        <p>
          v1.2.0-beta.17
        </p>
        <p>
          v1.2.0-beta.16
        </p>
        <p>
          v1.2.0-beta.15
        </p>
        <p>
          v1.2.0-beta.14
        </p>
        <p>
          v1.2.0-beta.13
        </p>
        <p>
          v1.2.0-beta.12
        </p>
        <p>
          v1.2.0-beta.11
        </p>
        <p>
          v1.2.0-beta.10
        </p>
        <p>
          v1.2.0-beta.9
        </p>
        <p>
          v1.2.0-beta.8
        </p>
        <p>
          v1.2.0-beta.7
        </p>
        <p>
          v1.2.0-beta.6
        </p>
        <p>
          v1.2.0-beta.5
        </p>
        <p>
          v1.2.0-beta.4
        </p>
        <p>
          v1.2.0-beta.3
        </p>
        <p>
          v1.2.0-beta.2
        </p>
        <p>
          v1.2.0-beta.1
        </p>
      </div>
    </div>
    <div
      class="octo-UI-c-dDCVUb"
      data-orientation="vertical"
      style="position: absolute; top: 0px; right: 0px; --radix-scroll-area-thumb-height: 18px;"
    />
    <div
      class="octo-UI-c-dDCVUb"
      data-orientation="horizontal"
      style="position: absolute; bottom: 0px; left: 0px; --radix-scroll-area-thumb-width: 18px;"
    />
  </div>
</div>
`;
